robot_skills/src/robot_skills/arms.py

```diff
@@ -861,40 +989,62 @@ def base_offset(self):
-        :param goal: tue_manipulation_msgs.msg.GraspPrecomputeGoal
-        :param color: list of rgb colors (0.0-1.0)
-        :param ns: namespace
-        :return: no return
-        """
-        marker = visualization_msgs.msg.Marker()
-        marker.header.frame_id = goal.goal.header.frame_id
-        marker.header.stamp = rospy.Time.now()
-        marker.type = 2
-        marker.pose.position.x = goal.goal.x
-        marker.pose.position.y = goal.goal.y
-        marker.pose.position.z = goal.goal.z
-        marker.lifetime = rospy.Duration(20.0)
-        marker.scale.x = 0.05
-        marker.scale.y = 0.05
-        marker.scale.z = 0.05
-        marker.ns = ns
+    def __repr__(self):
+        return "Arm(side='{side}')".format(side=self.side)
 
-        marker.color.a = 1
-        marker.color.r = color[0]
-        marker.color.g = color[1]
-        marker.color.b = color[2]
+def make_grasp_precompute_goal(frame_in_baselink, pre_grasp, first_joint_pos_only,
+                               allowed_touch_objects):
+    """
+    Construct a grasp precompute goal.
+
+    :param frame_in_baselink:
+    :param first_joint_pos_only: Bool to only execute first joint position of
+            whole trajectory.
+    :param allowed_touch_objects: List of object names in the worldmodel, which
+            are allowed to be touched.
+    :return: The constructed precompute goal.
+    """
+    # TODO: Get rid of this custom message type
+    # Create goal:
+    grasp_goal = GraspPrecomputeGoal()
+    grasp_goal.goal.header.frame_id = frame_in_baselink.frame_id
+    grasp_goal.goal.header.stamp = rospy.Time.now()
 
-        self._marker_publisher.publish(marker)
+    grasp_goal.PERFORM_PRE_GRASP = pre_grasp
+    grasp_goal.FIRST_JOINT_POS_ONLY = first_joint_pos_only
 
-    def __repr__(self):
-        return "Arm(side='{side}')".format(side=self.side)
+    grasp_goal.allowed_touch_objects = allowed_touch_objects
+
+    set_grasp_goal_position(grasp_goal, frame_in_baselink.frame.p)
+    set_grasp_goal_orientation(grasp_goal, frame_in_baselink.frame.M)
+    return grasp_goal
+
+def set_grasp_goal_position(grasp_goal, pos):
+    """
+    Set the position of the grasp goal.
+
+    :param grasp_goal: Grasp goal to change, modified in-place.
+    :param pos: Position to set.
+    """
+    grasp_goal.goal.x = pos.x()
+    grasp_goal.goal.y = pos.y()
+    grasp_goal.goal.z = pos.z()
+
+def set_grasp_goal_orientation(grasp_goal, ori):
+    """
+    Set the orientation of the grasp goal.
+
+    :param grasp_goal: Grasp goal to change, modified in-place.
+    :param ori: Orientation to set.
+    """
+    roll, pitch, yaw = ori.GetRPY()
+    grasp_goal.goal.roll  = roll
+    grasp_goal.goal.pitch = pitch
+    grasp_goal.goal.yaw   = yaw
 
 
 class ForceSensingArm(Arm):
-    def __init__(self, robot_name, tf_listener, get_joint_states, side):
+    def __init__(self, robot_name, tf_listener, get_joint_states, side, joints_piece=None):
```

Comment:
Zouden we dit dan niet ook anders moeten maken? Dus dat we b.v. een HeroArm hebben waar we al die pieces en een ForcePiece aan toevoegen?